native/wasmex/src/namespace.rs

```diff
@@ -12,7 +12,12 @@ use wasmer_runtime_core::{
 use crate::{atoms, instance::decode_function_param_terms};
 
 pub struct CallbackTokenResource {
-    pub token: (Mutex<Option<(bool, Vec<runtime::Value>)>>, Condvar),
+    pub token: (
+        Mutex<Option<(bool, Vec<runtime::Value>)>>,
+        Condvar,
+        Vec<String>,
```

Comment:
Passing string vectors into the token feels very un-rust-y - because we don't allow random strings, but only a known subset of strings. Sending over a `Vec<Type>` directly would save us some bytes and time wasted on converting these values. I only wonder whether `Vec<Type>` has copy'able (to satisfy the rust compiler because we move this data between threads) but maybe you can give it a try 🎱

Reply:
OK, using the `Type`s directly.